Very basic overview of Authz

bvhme opened the pull request:
For now, this is all that I could think of about Authz for now, for us to expand upon :)

Closes #24.

bvhme commented:
I'm also considering adding this one, however, that might be overkill

![oauth2 gopher 0 5x](https://user-images.githubusercontent.com/1917629/32457238-b91e2dda-c328-11e7-8732-aca3aef5f46c.png)

Cute though :) And I like designing these things.

evertlammerts commented on projects/authz.md:
```diff
@@ -0,0 +1,17 @@
+
+# Authz
+
+An implementation of the [OAuth 2.0 implicit authorization flow (RFC6749)](https://tools.ietf.org/html/rfc6749#section-1.3.2) as a HTTP service written in Go.
```
`An [OAuth 2.0](https://tools.ietf.org/html/rfc6749) compatible authorization service written in Go.`

evertlammerts commented on projects/authz.md:
```diff
@@ -0,0 +1,17 @@
+This service:
+* creates signed JSON Web Token (JWS) access tokens to be used what RFC6749 calls resource servers
+* Can use in-memory state storage or be connected to external storage such as Redis
+
```
```
This service:

* Implements the implicit flow as described in [RFC6749 section 4.2](https://tools.ietf.org/html/rfc6749#section-4.2)
* Uses JSON Web Keys ([RFC7517](https://tools.ietf.org/html/rfc7517)) for key management
* Creates JSON Web Tokens ([RFC7519](https://tools.ietf.org/html/rfc7519)) using HMAC or ECDSA (HS256, HS384, HS512, ES256, ES384, ES512)
* Provides interfaces for identity providers, state storage (with implementations for single node in-memory storage and Redis), authorization providers (to map users to scopes) and client registries.
```

evertlammerts commented:
De gopher vind ik wel leuk :-) Als je hem wilt publiceren moet er nog wel een 2.0 achter de OAuth.

bvhme commented:
Like this, without the text and a bit more bespoke?

![oauth2 gopher 0 25x](https://user-images.githubusercontent.com/1917629/32507580-19c34bba-c3e8-11e7-91c8-c8aef6c6f487.png)

evertlammerts commented:
I liked the first one better!

bvhme commented:
@evertlammerts I've added your changes, thanks very much.

And the OAuth logo is the same for 1 and 2, so I've added the most abstract official version there is :) I hope the Gopher is happy with its 👕 shirt

evertlammerts commented:
Thanks!
